### DISCUSSION TITLE
[help] Functions used in an Rmd that's rendered with tar_render() appear unused in tar_visnetwork() graph

### DISCUSSION THREAD

**Discussion by brndngrhm** [ropensci/targets#1361] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'll preface this with the fact that i'm relatively new to using targets. In my workflow one of the functions I use is called `get_totals()` and is defined in an R script called `helpers.R` which is sourced in my `_targets.R` file using the `tar_source()`. 

That function, `get_totals()`, is only used in an Rmd called `explore.Rmd` which is rendered as part of my pipeline using `tar_render()`. While I can see the `get_totals()` function in the interactive plot generated by `tar_visnetwork()`, it does not connect to anything and thus appears unused despite being used in `explore.Rmd`.

Is it possible for the plot created by `tar_visnetwork()` show functions used in documents rendered with `tar_render()`?

**Comment by wlandau** [comment, marked as answer]
For a `tar_render()` report, the only tracked dependencies are the upstream targets marked with `tar_load()` and/or `tar_read()`. If you find yourself using custom functions, it's usually better to create a few targets with those functions and then load them into the report, or if they are not used in any other targets, maybe include those functions in the report inline.

I don't have plans to change how `tar_render()` registers dependencies. One reason is to keep things consistent with `tar_quarto()`. Since Quarto runs in a separate process outside R, it does not automatically know about the global functions in a pipeline.
